packages/api/src/platforms/vtex/utils/shouldUpdateShippingData.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  IStoreGeoCoordinates,
+  IStoreSession,
+} from '../../../__generated__/schema'
+import {
+  CheckoutAddress,
+  OrderForm,
+  LogisticsInfo,
+} from '../clients/commerce/types/OrderForm'
+
+export const shouldUpdateShippingData = (
+  orderForm: OrderForm,
+  session: IStoreSession
+) => {
+  if (!hasItems(orderForm)) {
+    return false
+  }
+
+  if (!hasSessionPostalCodeOrGeoCoordinates(session)) {
+    return false
+  }
+
+  const { address, logisticsInfo } = orderForm.shippingData!
+
+  if (checkPostalCode(address, session.postalCode)) {
+    return true
+  }
+
+  if (checkGeoCoordinates(address, session.geoCoordinates)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryChannel(logisticsInfo, session)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryMethod(logisticsInfo, session)) {
+    return true
+  }
+
+  if (shouldUpdateDeliveryWindow(logisticsInfo, session)) {
+    return true
+  }
+  return false
+}
+
+// Validate if theres any item inside the orderForm
+const hasItems = (orderForm: OrderForm) => {
+  return orderForm.items.length !== 0
+}
+
+// Validate if theres any postal Code or GeoCoordinates set at the session
+const hasSessionPostalCodeOrGeoCoordinates = (session: IStoreSession) => {
+  return (
+    !!session.postalCode ||
+    (session.geoCoordinates &&
+      session.geoCoordinates.latitude &&
+      session.geoCoordinates.longitude)
+  )
+}
+
+// Validate if theres a difference between the session postal code and orderForm postal code
+const checkPostalCode = (
+  address: CheckoutAddress | null | undefined,
+  postalCode: string | null | undefined
+) => {
+  return typeof postalCode === 'string' && address?.postalCode !== postalCode
+}
+
+// Validate if theres a difference between the session geoCoords and orderForm geoCoords
+const checkGeoCoordinates = (
+  address: CheckoutAddress | null | undefined,
+  geoCoordinates: IStoreGeoCoordinates | null | undefined
+) => {
+  return (
+    typeof geoCoordinates?.latitude === 'number' &&
+    typeof geoCoordinates?.longitude === 'number' &&
+    (address?.geoCoordinates[0] !== geoCoordinates?.longitude ||
+      address?.geoCoordinates[1] !== geoCoordinates?.latitude)
+  )
+}
+
+// Validate if the deliveryChannel from the session is different from the selected delivery channel
+// and if so needs to validate if the deliveryChannel for the session is available inside the slas for the item
```

Comment:
Interesting. So if the session's `deliveryChannel` is unavailable for every item, it just doesn't update anything? Why is that?

Reply:
Yes!! It was the way I thought to avoid any unnecessary updates because the deliveryChannel will only change something if this sla is available for at least one item of the cart. So if you try to send a deliveryChannel that is not available for any item of the cart it would be the same as doing nothing because this type of delivery does not serve the cart at least at that moment.

Reply:
I see. Why shouldn't we change it and let the user deal with the consequences? I'm afraid the user might perceive the behavior as the website "not working", because it can't do what it wants. What would happen if we did change even if there are no items with available sla?

Reply:
In my view it makes no sense since sending this information when no item has this sla will not generate absolutely no changes at the orderForm level. So we would just be generating completely unnecessary calls on our side every time validateCart was called because this condition would never change. In addition, excessive calls in this sense can have a bad side effect on the orderForm, which is to rewrite the orderForm during the user's action on the site, causing items to "disappear", for example, because the call occurs at the same time as the user's action. Remember that validateCart runs whenever the customer adds an item to the cart, for example, so imagine a PLP with an add cart button and the customer adds 10 items (none of them have that sla set in the available session) and you would be generating 10 calls to update shipping information that don't change anything in the orderform, but as I mentioned it can generate strange side effects in the navigation.

Reply:
So, if I got it right, it's like the orderform has this same internal logic when we ask for it to update. Meaning that if I send a `deliveryChannel` that none of the items have an SLA for, nothing changes in the orderform. Is this right? Or it does change internally, it just doesn't have any effect on price or availability or etc?

Because if I'm wrong and the orderform does change internally, then I think it's worth sending the change anyway, as we'd be reflecting the orderform behavior. If this is the case, in your example, the shipping information would be changed only when needed, since the orderform would already have incorporated the change previously, so our diffing algorithms would not detect a change.

Reply:
Documented created: https://miro.com/app/board/uXjVM8Yftrw=/?share_link_id=901261295826

Just to document it here also:
After some tests using checkout APIs my conclusion is that it would make sense to update always when its a change considering the cart as a whole (zipCode or geoCoords) but changes that affect itens wont cause any structural change at the orderForm so it´s better to keep the logic implemented to avoid those calls.